### PR TITLE
Setup proxy server to render all components

### PR DESCRIPTION
**Pull request by mchung1128**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "devDependencies": {
+    "@babel/core": "^7.5.5",
+    "@babel/preset-env": "^7.5.5",
+    "@babel/preset-react": "^7.0.0",
+    "babel-loader": "^8.0.6",
+    "webpack": "^4.36.1",
+    "webpack-cli": "^3.3.6"
+  }
+}
```
Comment (matt-leyrat) [inline package.json]: Looks about right!

File: package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "mike-proxy",
+  "version": "1.0.0",
+  "description": "Proxy server for fec3-galadriel",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start:dev": "nodemon server/server.js",
```
Comment (matt-leyrat) [inline package.json]: For future reference, you can just point it at your server directory and it'll run index.js automatically afaik.

File: public/runtime.bundle.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
+/******/ (function(modules) { // webpackBootstrap
```
Comment (matt-leyrat) [inline public/runtime.bundle.js]: I don't think we should be committing our bundles

File: server/server.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const photos = 'http://localhost:3001';
+const itemSummary = 'http://localhost:3002';
+const relatedProducts = 'http://localhost:3003';
+const reviews = 'http://localhost:3004';
+
+
+app.use('/:id', express.static('./public'));
+
+app.all('/api/photos/:id', (req, res) => {
+  console.log('Connected to Photos!');
```
Comment (matt-leyrat) [inline server/server.js]: I wonder if these logs should go after the proxy.web() invocation. Seems like they're preemptively logging a success before the attempt is made to connect

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        }
+      }
+    }
+  },
+  resolve: {
+    alias: {
+      "styled-components": path.resolve(__dirname, "node_modules", "styled-components")
+    }
+  }
+};
```
Comment (matt-leyrat) [inline webpack.config.js]: LGTM, just remember to add a new line at the end of your files :) 

Adding a linter to the proxy would make it easier in the long run, although it seems like you have most of it setup by now.

**Review by matt-leyrat — APPROVED** [review]
Looks ready to merge 👍

**Comment by garrowat** [comment]
Looks great, nice work!

**Pull request merged**